cmd/signozcollector/main.go

```diff
@@ -1,43 +1,96 @@
 package main
 
 import (
+	"context"
 	"fmt"
 	"log"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
 
-	"github.com/SigNoz/signoz-otel-collector/components"
-	"go.opentelemetry.io/collector/component"
-	"go.opentelemetry.io/collector/service"
+	"github.com/SigNoz/signoz-otel-collector/service"
+	"github.com/SigNoz/signoz-otel-collector/signozcol"
+	flag "github.com/spf13/pflag"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
 )
 
 func main() {
 
-	factories, err := components.Components()
+	// Command line flags
+	f := flag.NewFlagSet("Collector CLI Options", flag.ExitOnError)
+
+	f.Usage = func() {
+		fmt.Println(f.FlagUsages())
+		os.Exit(0)
+	}
+
+	f.String("config", "", "File path for the collector configuration")
+	f.String("manager-config", "", "File path for the agent manager configuration")
+	err := f.Parse(os.Args[1:])
 	if err != nil {
-		log.Fatalf("failed to build default components: %v", err)
+		log.Fatalf("Failed to parse args %v", err)
 	}
 
-	info := component.BuildInfo{
-		Command:     "signoz-otel-collector",
-		Description: "SigNoz OTEL Collector",
-		Version:     "latest",
+	collectorConfig, _ := f.GetString("config")
+	managerConfig, _ := f.GetString("manager-config")
+
+	// logger, err := zap.NewProduction()
+	logger, err := initZapLog()
+	if err != nil {
+		log.Fatalf("failed to initialize zap logger: %v", err)
 	}
 
-	params := service.CollectorSettings{
-		Factories: factories,
-		BuildInfo: info,
+	ctx := context.Background()
+
+	coll := signozcol.New(
+		signozcol.WrappedCollectorSettings{
+			ConfigPaths: []string{collectorConfig},
+			// TODO: Build version from git tag
+			Version:     "0.63.0",
```

Comment:
@srikanthccv  do we need to add this now?

Reply:
Can you please add then changes? we can merge it after that.

Reply:
The deployment patten is not yet clear in k8s, so I would hold it till we figure it out.

Reply:
Got it